misc/futures-bounded/src/unique.rs

```diff
@@ -0,0 +1,230 @@
+use std::collections::HashMap;
+use std::future::Future;
+use std::hash::Hash;
+use std::task::{Context, Poll, Waker};
+use std::time::Duration;
+
+use futures_timer::Delay;
+use futures_util::future::BoxFuture;
+use futures_util::FutureExt;
+
+use crate::Timeout;
+
+/// Represents a set of (Worker)-[Future]s.
+///
+/// Each worker must finish within the specified time and the set never outgrows its capacity.
+pub struct UniqueWorkers<ID, O> {
+    timeout: Duration,
+    capacity: usize,
+    inner: HashMap<ID, (BoxFuture<'static, O>, Delay)>,
+    empty_waker: Option<Waker>,
+    full_waker: Option<Waker>,
+}
+
+/// Error of a worker pushing
+#[derive(PartialEq, Debug)]
+pub enum PushError<F> {
+    /// The length of the set is equal to the capacity
+    BeyondCapacity(F),
+    /// The set already contains the given worker's ID
+    ReplacedWorker(F),
+}
+
+impl<ID, O> UniqueWorkers<ID, O> {
+    pub fn new(timeout: Duration, capacity: usize) -> Self {
+        Self {
+            timeout,
+            capacity,
+            inner: Default::default(),
+            empty_waker: None,
+            full_waker: None,
+        }
+    }
+}
+
+impl<ID, O> UniqueWorkers<ID, O>
+where
+    ID: Clone + Hash + Eq + Send + 'static,
+{
+    /// Push a worker into the set.
+    /// This method adds the given worker with defined `worker_id` to the set.
```

Comment:
```suggestion
    /// Push a worker into the set.
    ///
    /// This method adds the given worker with defined `worker_id` to the set.
```

By convention, rustdoc has a newline here :)